geomesa-core/src/main/scala/org/locationtech/geomesa/core/csv/package.scala

```diff
@@ -39,84 +39,80 @@ import scala.collection.JavaConversions._
 import scala.io.Source
 import scala.util.Success
 
-case class TypeSchema(name: String, schema: String)
+case class TypeSchema(name: String, schema: String, latLonFields: Option[(String, String)])
+
 package object csv extends Logging {
 
+  def guessTypes(csvFile: File, hasHeader: Boolean): TypeSchema = {
+    val typename = FilenameUtils.getBaseName(csvFile.getName)
+    val reader = Source.fromFile(csvFile).bufferedReader()
+    val guess = guessTypes(typename, reader, hasHeader)
+    reader.close()
+    guess
+  }
 
-  // this can probably be cleaned up and simplified now that parsers don't need to do double duty...
-  def typeData(rawData: TraversableOnce[String]): Seq[Char] = {
-    def tryAllParsers(datum: String): Char =
-      CSVParser.parsers.view.map(_.parseAndType(datum)).collectFirst { case Success(x) => x } match {
-        case Some(x) => x._2
-        case None    => 's'   // should get to this anyway as StringParser is guaranteed to succeed
-      }
+  def csvToFeatures(csvFile: File,
+                    hasHeader: Boolean,
+                    typeSchema: TypeSchema): SimpleFeatureCollection = {
+    val sft = SimpleFeatureTypes.createType(typeSchema.name, typeSchema.schema)
+    buildFeatureCollection(csvFile, hasHeader, sft, typeSchema.latLonFields)
+  }
 
+  protected[csv] def tryParsers(rawData: TraversableOnce[String]): Seq[CSVParser[_]] = {
+    def tryAllParsers(datum: String) =
+      CSVParser.parsers.find(_.parse(datum).isSuccess).getOrElse(StringParser)
     rawData.map(tryAllParsers).toSeq
   }
-  
-  def sampleRecords(records: jIterator[CSVRecord], hasHeader: Boolean): (Seq[String], CSVRecord) =
-    if (hasHeader) {
-      val header = records.next
-      val record = records.next
-      (header.toSeq, record)
-    } else {
-      val record = records.next
-      val header = Seq.tabulate(record.size()) { n => s"C$n" }
-      (header, record)
-    }
 
-  def guessTypes(name: String,
+  protected[csv] def guessHeaders(record: CSVRecord, hasHeader: Boolean = true): Seq[String] =
+    if (hasHeader) record.toSeq else Seq.tabulate(record.size())(n => s"C$n")
+
+  protected[csv] def guessTypes(name: String,
                  csvReader: Reader,
                  hasHeader: Boolean = true,
-                 format: CSVFormat = CSVFormat.DEFAULT): TypeSchema = {
-    val records = format.parse(csvReader).iterator
-    val (header, record) = sampleRecords(records, hasHeader)
-    val typeChars = typeData(record.iterator)
+                 format: CSVFormat = CSVFormat.DEFAULT,
+                 numSamples: Int = 5): TypeSchema = {
+    assert(numSamples > 0)
+    val records = format.parse(csvReader).iterator.take(numSamples + 1).toSeq
+    assert(records.size > 1 || (!hasHeader && records.size > 0))
+
+    val headers = guessHeaders(records(0), hasHeader)
+    val sample = records.drop(1)
```

Comment:
This is just for guessing the types, so it shouldn't matter that we drop the header - we just want a few rows we can look at. It would make sense to use the 'hasHeader' flag, but if somehow there was a header when we thought there wasn't, the type inference wouldn't work. So, I can see it both ways...
What do you think?